azurerm/resource_arm_api_management_logger.go

```diff
@@ -0,0 +1,256 @@
+
+	eventHubRaw := d.Get("eventhub").([]interface{})
+	appInsightsRaw := d.Get("application_insights").([]interface{})
+
+	if len(eventHubRaw) == 0 && len(appInsightsRaw) == 0 {
+		return fmt.Errorf("Either `eventhub` or `application_insights` is required")
+	}
+
+	parameters := apimanagement.LoggerContract{
+		LoggerContractProperties: &apimanagement.LoggerContractProperties{
+			IsBuffered:  utils.Bool(d.Get("buffered").(bool)),
+			Description: utils.String(d.Get("description").(string)),
+		},
+	}
+
+	if len(eventHubRaw) > 0 {
+		parameters.LoggerType = apimanagement.AzureEventHub
+		parameters.Credentials = expandArmApiManagementLoggerEventHub(eventHubRaw)
+	} else if len(appInsightsRaw) > 0 {
+		parameters.LoggerType = apimanagement.ApplicationInsights
+		parameters.Credentials = expandArmApiManagementLoggerApplicationInsights(appInsightsRaw)
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, serviceName, name, parameters, ""); err != nil {
+		return fmt.Errorf("Error creating Logger %q (Resource Group %q / API Management Service %q): %+v", name, resourceGroup, serviceName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, serviceName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Logger %q (Resource Group %q / API Management Service %q): %+v", name, resourceGroup, serviceName, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Logger %q (Resource Group %q / API Management Service %q) ID", name, resourceGroup, serviceName)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmApiManagementLoggerRead(d, meta)
+}
+
+func resourceArmApiManagementLoggerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).apiManagementLoggerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	serviceName := id.Path["service"]
+	name := id.Path["loggers"]
+
+	resp, err := client.Get(ctx, resourceGroup, serviceName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Logger %q (API Management Service %q / Resource Group %q) was not found - removing from state", name, serviceName, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Logger %q (API Management Service %q / Resource Group %q): %+v", name, serviceName, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("api_management_name", serviceName)
+
+	if properties := resp.LoggerContractProperties; properties != nil {
+		d.Set("buffered", properties.IsBuffered)
+		d.Set("description", properties.Description)
+		if err := d.Set("eventhub", flattenArmApiManagementLoggerEventHub(d, properties)); err != nil {
+			return fmt.Errorf("Error setting `eventhub` for Logger %q (Resource Group %q / API Management Name %q): %s", name, resourceGroup, serviceName, err)
```

Comment:
we can update this to:

```suggestion
			return fmt.Errorf("Error setting `eventhub`: %s", err)
```